67_RayQueryGeometry/app_resources/render.comp.hlsl

```diff
@@ -19,14 +21,12 @@ cbuffer CameraData
 
 [[vk::binding(2, 0)]] RWTexture2D<float4> outImage;
 
-// TODO: don't know if this is correct
-float3 getNormalsFromMask(int bn, int mask)
+float3 getNormalsFromMask(uint32_t n)
 {
-    float3 n = (float3)0;
-    n.x = float(asint((bn >> 24) & mask));
-    n.y = float((bn >> 16) & mask);
-    n.z = float((bn >> 8) & mask);
-    return n;
+    // this still doesn't feel right
+    // int8_t3 --> SSCALED (converts to float) --> unpackUnorm (read as uint --> float / 255)
+    float4 v = nbl::hlsl::spirv::unpackUnorm4x8(n) * 255.0;
+    return v.xyz;
```

Comment:
the encoding scheme of the normal is kinda funny, you just need to normalize it instead of multiplying by any constant

Reply:
there's also an `unpackSnorm4x8` and that will give you `[-127,127]` range